rescue/provisioner_rescue.go

```
@@ -0,0 +1,83 @@
+package rescue
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/src-d/terraform-provider-online-net/online"
+)
+
+var provisionerOnlineClient online.Client
+
+func applyRescueProvisioner(ctx context.Context) error {
+	d := ctx.Value(schema.ProvConfigDataKey).(*schema.ResourceData)
+
+	if provisionerOnlineClient == nil {
+		token := d.Get("token").(string)
+		provisionerOnlineClient = online.NewClient(token)
+	}
+
+	enabled := d.Get("enabled").(bool)
+	server := d.Get("server").(int)
+	imageInterface, imageExists := d.GetOk("image")
+	pathInterface, pathExists := d.GetOk("credentials_path")
+
+	if enabled && !imageExists {
+		return errors.New("Need image to enable rescue mode")
+	}
+
+	if enabled && !pathExists {
+		return errors.New("Need credentials_path to enable rescue mode")
+	}
+
+	if enabled {
+		credentials, err := provisionerOnlineClient.BootRescueMode(server, imageInterface.(string))
+		if err != nil {
+			return err
+		}
+		err = writeCredentials(credentials, pathInterface.(string))
+		if err != nil {
+			return err
+		}
+	} else {
+		err := provisionerOnlineClient.BootNormalMode(server)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// writeCredentials is a hack as provisioners can not have output data.
+// This will write the credentials to individual files that can be picked up by ${file()}
+func writeCredentials(credentials *online.RescueCredentials, to string) error {
+
+	os.Mkdir(to, 0755)
```

Why error is not handled here?

---

To not error when the directory already exists

---

But you have to handle the case where there's a real error, e.g. you cannot write in the specified path...

---

Good point, will add a marcher on the error